services/app-api/src/resolvers/healthPlanPackage/updateHealthPlanFormData.ts

```diff
@@ -59,168 +114,277 @@ export function updateHealthPlanFormDataResolver(
         }
 
         const unlockedFormData: UnlockedHealthPlanFormDataType = formDataResult
-        const result = await store.findHealthPlanPackage(input.pkgID)
 
-        if (isStoreError(result)) {
-            console.info('Error finding a package', result)
-            const errMessage = `Issue finding a package of type ${result.code}. Message: ${result.message}`
-            logError('updateHealthPlanFormData', errMessage)
-            setErrorAttributesOnActiveSpan(errMessage, span)
-            throw new Error(errMessage)
-        }
+        // Uses new DB if flag is on
+        if (ratesDatabaseRefactor) {
+            // Find contract from DB
+            const contractWithHistory = await store.findContractWithHistory(
+                input.pkgID
+            )
 
-        if (result === undefined) {
-            const errMessage = `No package found to update with that ID: ${input.pkgID}`
-            logError('updateHealthPlanFormData', errMessage)
-            setErrorAttributesOnActiveSpan(errMessage, span)
-            throw new UserInputError(errMessage, {
-                argumentName: 'pgkID',
-            })
-        }
+            if (contractWithHistory instanceof Error) {
+                const errMessage = `Issue finding a contract with history with id ${input.pkgID}. Message: ${contractWithHistory.message}`
+                logError('fetchHealthPlanPackage', errMessage)
+                setErrorAttributesOnActiveSpan(errMessage, span)
 
-        const planPackage: HealthPlanPackageType = result
+                if (contractWithHistory instanceof NotFoundError) {
+                    throw new GraphQLError(errMessage, {
+                        extensions: {
+                            code: 'NOT_FOUND',
+                            cause: 'DB_ERROR',
```

Comment:
does every error need a cause? I'm not sure it makes sense to me to call a not found a db error, what other causes do we use?

Reply:
It's not required to have a `cause`. I've been using it to specify where this error happened. Some other causes: `PROTO_DECODE_ERROR`, `EMAIL_ERROR`, `INVALID_PACKAGE_STATUS`.